### ISSUE TITLE
settings.conf key not working, stays in "Evaluation Version"

### ISSUE THREAD

**Issue by jcfrosty** [rix1337/docker-ripper#95]
Have you read and understood this [part of the readme](https://github.com/rix1337/docker-ripper#do-you-offer-support)?

[X] YES

### Issue description:

I do indeed have a KEY. I've purchased the full version, renamed the settings.conf and it IS recognized, but still runs MakeMKV as "Evaluation version". The Docker Log shows the init script is locating the settings.conf file and copying it to the proper location.

```
*** Running /etc/my_init.d/10_syslog-ng.init...
Apr  4 22:13:25 Oceans syslog-ng[13]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/ripper.sh...
Using this daily? Please sponsor me at https://github.com/sponsors/rix1337 - any amount counts!
Found settings.conf. Replacing beta key file.
*** Running /etc/my_init.d/web.sh...
Starting web ui
*** Booting runit daemon...
*** Runit started as PID 38
04.04.2022 22:13:27 : Starting Ripper. Optical Discs will be detected and ripped within 60 seconds.
Apr  4 22:13:27 Oceans cron[42]: (CRON) INFO (pidfile fd = 3)
Apr  4 22:13:27 Oceans cron[42]: (CRON) INFO (Running @reboot jobs)
04.04.2022 22:13:27 : Unexpected makemkvcon output: DRV:0,256,999,0,"","",""
*** Shutting down runit daemon (PID 38)...
*** Running /etc/my_init.post_shutdown.d/10_syslog-ng.shutdown...
Apr  4 22:14:09 Oceans syslog-ng[13]: syslog-ng shutting down; version='3.25.1'
*** Init system aborted.
*** Killing all processes...
*** Running /etc/my_init.d/10_syslog-ng.init...
Apr  4 22:17:26 Oceans syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/ripper.sh...
Using this daily? Please sponsor me at https://github.com/sponsors/rix1337 - any amount counts!
Found settings.conf. Replacing beta key file.
*** Running /etc/my_init.d/web.sh...
Starting web ui
*** Booting runit daemon...
*** Runit started as PID 37
04.04.2022 22:17:28 : Starting Ripper. Optical Discs will be detected and ripped within 60 seconds.
Apr  4 22:17:28 Oceans cron[41]: (CRON) INFO (pidfile fd = 3)
Apr  4 22:17:28 Oceans cron[41]: (CRON) INFO (Skipping @reboot jobs -- not system startup)
04.04.2022 22:17:28 : Unexpected makemkvcon output: DRV:0,256,999,0,"","",""
04.04.2022 22:18:29 : Unexpected makemkvcon output: DRV:0,256,999,0,"","",""
*** Shutting down runit daemon (PID 37)...
*** Running /etc/my_init.post_shutdown.d/10_syslog-ng.shutdown...
Apr  4 22:19:12 Oceans syslog-ng[12]: syslog-ng shutting down; version='3.25.1'
*** Init system aborted.
*** Killing all processes...
*** Running /etc/my_init.d/10_syslog-ng.init...
Apr  4 22:20:24 Oceans syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/ripper.sh...
Using this daily? Please sponsor me at https://github.com/sponsors/rix1337 - any amount counts!
Found settings.conf. Replacing beta key file.
*** Running /etc/my_init.d/web.sh...
Starting web ui
04.04.2022 22:20:26 : Starting Ripper. Optical Discs will be detected and ripped within 60 seconds.
*** Booting runit daemon...
*** Runit started as PID 37
Apr  4 22:20:26 Oceans cron[42]: (CRON) INFO (pidfile fd = 3)
Apr  4 22:20:26 Oceans cron[42]: (CRON) INFO (Skipping @reboot jobs -- not system startup)
04.04.2022 22:20:26 : Disk tray open
04.04.2022 22:21:32 : BluRay detected: Saving MKV
```

As you can see, its still running in eval mode.

```
MSG:1005,0,1,"MakeMKV v1.16.7 linux(x64-release) started","%1 started","MakeMKV v1.16.7 linux(x64-release)"
MSG:5074,0,0,"Automatic checking for updates is enabled, you may disable it in preferences if you don't want MakeMKV to contact web server.","Automatic checking for updates is enabled, you may disable it in preferences if you don't want MakeMKV to contact web server."
DRV:0,2,999,28,"BD-RE HL-DT-ST BD-RE  WH16NS60 1.02 M00LBCB3515","LIVE_FREE_DIE_HARD","/dev/sr0"
DRV:1,256,999,0,"","",""
DRV:2,256,999,0,"","",""
DRV:3,256,999,0,"","",""
DRV:4,256,999,0,"","",""
DRV:5,256,999,0,"","",""
DRV:6,256,999,0,"","",""
DRV:7,256,999,0,"","",""
DRV:8,256,999,0,"","",""
DRV:9,256,999,0,"","",""
DRV:10,256,999,0,"","",""
DRV:11,256,999,0,"","",""
DRV:12,256,999,0,"","",""
DRV:13,256,999,0,"","",""
DRV:14,256,999,0,"","",""
DRV:15,256,999,0,"","",""
MSG:1011,0,1,"Using LibreDrive mode (v06.3 id=7FA2657C7667)","%1","Using LibreDrive mode (v06.3 id=7FA2657C7667)"
MSG:3007,0,0,"Using direct disc access mode","Using direct disc access mode"
MSG:5053,1544,1,"This functionality is shareware. You may evaluate it for 30 days after what you would need to purchase an activation key if you like the functionality. Do you want to start evaluation period now?","This functionality is shareware. You may evaluate it for %1 days after what you would need to purchase an activation key if you like the functionality. Do you want to start evaluation period now?","30"
MSG:5050,0,2,"Evaluation version, 30 day(s) out of 30 remaining","Evaluation version, %1 day(s) out of %2 remaining","30","30"
```

**Comment by rix1337** [comment]
Hi @jcfrosty ,
your issue seems related to #93 
Could you please run `makemkvcon reg` within your container and report back if that helps?

People on the other issue were unable to confirm this fix.

If the message on your end disappears after running that command, I'll include it after copying settings.conf to the final location.

**Comment by rix1337** [comment]
Not owning a MakeMKV key, I can not test/confirm this myself.

Any feedback helps.

**Comment by edgd1er** [comment]
@jcfrosty If the makemkvcon is not displaying a registered status, then /root/.Makemkv/settings.conf is missing for any reason.

Look around here, this could be where is the solution to your problem:
https://github.com/rix1337/docker-ripper/blob/5ff7f7bbc4c237f734c67a3b2da1048c6d6ef192/root/etc/my_init.d/ripper.sh#L24

I would have written something like: 
- elif [ "$KEY# != "" ] ...
- elif [ -n $KEY ] ...

PS: even with random serial, makemkv will find /root/.Makemkv/settings.conf, and complain about it being invalid.

**Comment by rix1337** [comment]
Thanks @edgd1er 
Could you guys try out the latest build? I have made small tweaks as recommended.
https://github.com/rix1337/docker-ripper/commit/6a9663e026d75771f26fd9da76665095ebabb70a

**Comment by edgd1er** [comment]
`makemkvcon reg`does not display the current validated key, I would replace the line with `makemkvcon reg "" 2>/dev/null`  in order to display the current key, knowing that would appear in the logs and is not very safe to keep the serial private.

I didn't find a command that would confirm that the programm is registered.

**Comment by jcfrosty** [comment]
Sorry, I'll give it a test. I've been troubleshooting a udev issue that I recently found how to fix. It kept locking up the /dev/sr0.

EDIT:
It appears the notice will not show up when ripping DVDs as thats a freeware function. It only shows up when ripping Blurays which is a shareware function.

```
MSG:5050,0,2,"Evaluation version, 30 day(s) out of 30 remaining","Evaluation version, %1 day(s) out of %2 remaining","30","30"
--
MSG:5053,1544,1,"This functionality is shareware. You may evaluate it for 30 days after what you would need to purchase an activation key if you like the functionality. Do you want to start evaluation period now?","This functionality is shareware. You may evaluate it for %1 days after what you would need to purchase an activation key if you like the functionality. Do you want to start evaluation period now?","30"
MSG:3007,0,0,"Using direct disc access mode","Using direct disc access mode"
```
EDIT2: I've tried copying my settings file from my desktop which uses the registered version and that doesn't appear to have worked either.

**Comment by davidlambjr** [comment]
I ran into this issue as well and after much digging and playing, I found that the ripper.sh script is running as nobody, not root, so it isn't getting the settings file. changing the line that calls the ripper.sh script to: "setuser root /ripper/ripper.sh &" resolved the issue for me. Probably not the best solution, as the script is running as root now, but it is able to grab the settings.conf file correctly.
